persistence/sql/sql_process_store.go

```diff
@@ -269,6 +269,12 @@ func (p sqlProcessStoreImpl) applyTerminateIfRunningPolicy(
 		return nil, err
 	}
 
+	// lock process execution row first
+	_, err = tx.SelectProcessExecutionForUpdate(ctx, latestProcessExecution.ProcessExecutionId)
```

Comment:
How about just replace line 281 with this? 
Because the else(not found) branch won't update anything for this execution so it's still a bit wasteful to lock it.

Reply:
right, not found case is just creating states for a new process